[AMDGPU] Clean up functions for checking inline literals

shiltian opened the pull request:
This patch cleans up functions for checking inline literals.

arsenm commented on llvm/lib/Target/AMDGPU/AMDGPUISelDAGToDAG.h:
```diff
@@ -107,10 +107,6 @@ class AMDGPUDAGToDAGISel : public SelectionDAGISel {
   std::pair<SDValue, SDValue> foldFrameIndex(SDValue N) const;
   bool isInlineImmediate(const SDNode *N) const;
 
-  bool isInlineImmediate16(int64_t Imm) const {
```
What's the plus of removing this if there's no change to the underlying helper function?

shiltian replied:
The change of `AMDGPU::isInlinableLiteral16` will come soon and then there is no sufficient information here to call the new function. Besides, these functions are literally unused. I don't see any point of keeping them.

Sisyph commented on llvm/lib/Target/AMDGPU/AMDGPUInstructionSelector.h:
```diff
@@ -353,7 +353,6 @@ class AMDGPUInstructionSelector final : public InstructionSelector {
   void renderFPPow2ToExponent(MachineInstrBuilder &MIB, const MachineInstr &MI,
                               int OpIdx) const;
 
-  bool isInlineImmediate16(int64_t Imm) const;
```
I see this function as well as isInlineImmediate32 and isInlineImmediate64 are called in
AMDGPUGenGlobalISel.inc AMDGPUInstructionSelector::testImmPredicate_I64, but I'm not sure if that function is used. 

Perhaps you can remove all 3 functions?

shiltian commented:
gentle ping

jayfoad commented on llvm/lib/Target/AMDGPU/SIInstrInfo.td:
```diff
@@ -797,16 +797,8 @@ def i64imm_32bit : ImmLeaf<i64, [{
-
 def InlineImm64 : ImmLeaf<i64, [{
-  return isInlineImmediate64(Imm);
+  return isInlineImmediate(APInt(64, static_cast<uint64_t>(Imm)));
```
Can you use IntImmLeaf instead of ImmLeaf here? That way Imm will already be an APInt so you won't have to construct a new APInt.

jayfoad reviewed (approved):
LGTM.

shiltian committed:
[AMDGPU] Remove unused functions for checking 16-bit inline literals

This patch removes unused functions that check if an immediate is a 16-bit inline
literals. This serves as prime patches to fix #79369.
